collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormsDetailsFetcher.java

```diff
@@ -158,4 +160,16 @@ private static boolean isMediaFileAlreadyDownloaded(List<File> localMediaFiles,
     private String getMd5HashWithoutPrefix(String hash) {
         return hash == null || hash.isEmpty() ? null : hash.substring("md5:".length());
     }
+
+    private Form getFormByVersion(List<Form> forms, @Nullable String expectedVersion) {
+        for (Form form : forms) {
+            if (expectedVersion == null && form.getVersion() == null) {
```

Comment:
You could use `Objects.equals(form.getVersion(), expectedVersion)`, I believe.

Reply:
Right, fixed.